graspologic/nominate/spectralVN.py

```diff
@@ -0,0 +1,300 @@
+                raise IndexError("Adjacency matrix or embedding must have dim 2")
+            elif not self.input_graph_:
+                # embedding was provided
+                if X.shape[1] > X.shape[0]:
+                    raise IndexError(
+                        "dim 1 of an embedding should be smaller than dim 0."
+                    )
+            elif X.shape[0] != X.shape[1]:
+                raise IndexError("Adjacency Matrix should be square.")
+        else:
+            raise NotImplementedError("Multigraph SVN not implemented")
+        # check y
+        if not isinstance(y, np.ndarray):
+            raise TypeError("y must be of type np.ndarray")
+        elif not np.issubdtype(y.dtype, np.integer):
+            raise TypeError("y must have dtype int")
+        elif np.ndim(y) > 2 or (y.ndim == 2 and y.shape[1] > 2):
+            raise IndexError("y must have shape (n) or (n, 1) or (n, 2).")
+        # check k
+        if k is not None and type(k) is not int:
+            raise TypeError("k must be an integer")
+        elif k is not None and k <= 0:
+            raise ValueError("k must be greater than 0")
+
+        if self.input_graph_:
+            if not isinstance(self.embedder_, BaseSpectralEmbed) and not isinstance(
+                self.embedder_, str
+            ):
+                raise TypeError(
+                    "embedder must be either of type str or BaseSpectralEmbed"
+                )
+
+    def _embed(self, X: np.ndarray):
+        # Embed graph if embedding not provided
+        if self.input_graph_:
+            if isinstance(self.embedder_, BaseSpectralEmbed):
+                embedder = self.embedder_
+            elif self.embedder_ == "ASE":
+                embedder = ase()
+            elif self.embedder_ == "LSE":
+                embedder = lse()
+            else:
+                raise ValueError(
+                    "Requested embedding method does not exist, if str is passed must "
+                    "be either 'ASE' or 'LSE'."
+                )
+            self.embedding_ = embedder.fit_transform(X)
+        else:
+            self.embedding_ = X
+
+    def fit(
+        self,
+        X: np.ndarray,
+        y: np.ndarray,
+        k: int = None,
+        metric: str = "euclidean",
+        metric_params: dict = None,
+    ):
+        """
+        Constructs the embedding if not provided, then calculates the pairwise distance
+        from each seed to each vertex in graph.
+
+        Parameters
+        ----------
+        X : np.ndarray
+            - If `input_graph` is True
+                Expects a graph as an adjacency matrix, i.e. an ndarray of shape (n, n).
+                Will be embedded using the specified embedder.
+            - If `input_graph` is False
+                Expects an embedding of the graph, i.e. a ndarray of size (n, d).
+        y: np.ndarray
+            List of seed vertex indices, OR List of tuples of seed vertex indices and
```

Comment:
i think you should probably clarify the language, seems like what you're expecting isn't exactly a list of tuples